src/login/logind-session.c

```diff
@@ -529,6 +534,14 @@ int session_load(Session *s) {
                         s->type = t;
         }
 
+        if (original_type) {
+                SessionType ot;
+
+                ot = session_type_from_string(original_type);
+                if (ot >= 0)
+                        s->original_type = ot;
+        }
```

Comment:
let#s do some basic compat when upgrading, i.e. add an else branch here, and initialize the original type from the current type if the `ORIGINAL_TYPE` field is not set in the state file